Refs QA-3628 Consistent use of italic blank for empty strings...

kmtracey opened the pull request:
…and wrap all-whitespace in quotes.

(This is in the display (non-edit view) of the key/values for the key/value widget.)

## Product Description
<!-- For non-invisible changes, describe user-facing effects. -->

## Technical Summary
<!--
    Provide a link to the ticket or document which prompted this change,
    Describe the rationale and design decisions.
-->

## Feature Flag
<!-- If this is specific to a feature flag, which one? -->

## Safety Assurance

### Safety story
<!--
Describe how you became confident in this change, such as
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.
-->

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->


### Migrations
<!-- Delete this section if the PR does not contain any migrations -->
- [ ] The migrations in this code can be safely applied first independently of the code

<!-- Please link to any past code changes that are coordinated with this migration -->

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [ ] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [ ] Risk label is set correctly
- [ ] The set of people pinged as reviewers is appropriate for the level of risk of the change

kmtracey committed:
Refs QA-3628 Consistent use of italic blank for empty strings and wrap all-whitespace in quotes.

(This is in the display (non-edit view) of the key/values for the key/value widget.)

orangejenny reviewed (changes requested):
Just need to make sure the user-input values are sanitized, otherwise this looks good.

kmtracey commented on corehq/apps/hqwebapp/static/hqwebapp/js/ui_elements/ui-element-input-map.js:
```diff
@@ -81,12 +81,11 @@ hqDefine('hqwebapp/js/ui_elements/ui-element-input-map', [
-                this.$noedit_view.html(left_side + ' &rarr; ' + (
-                    map_val ? $('<div>').text(map_val).html() : django.gettext('(empty)')
-                ));
+                if (map_key && !map_key.trim()) map_key = `"<span style="white-space: pre;">${map_key}</span>"`;
```
Ah, oof. If I use `DOMPurify.sanitize` that strips the offending content vs what had been there escapes it. I like the clarity of `DOMPurify` (since obviously the point of the successive `.text` `.html` jquery calls passed me by) but I can't in a brief look find a way to make `DOMPurify` simply escape rather than strip...is it really OK to strip out offending content or is there way I'm missing to make `DOMPurify` escape?

orangejenny replied:
I think it's fine to strip any tags - I can't imagine anyone was relying on the behavior of showing the escaped HTML.

kmtracey replied:
OK I pushed up a change that uses `DOMPurify`. I did it in a way that fully-stripped values get turned into the italic blank, not sure if that is best or not. (The alternative is fully-stripped values just appear not to be there leaving possibly a lone arrow. Neither seems particularly great but perhaps this is an extreme edge case?)

orangejenny replied:
I think it's an edgy enough edge case not to worry about.

kmtracey committed:
Use DOMPurify to sanitize key/val for display.

(Also adjust formatting per sticklerbot, to some extent...not to the extent of changing variable name patterns in this code.)
